manifests/javagateway.pp

```diff
@@ -24,6 +24,11 @@
   $startpollers             = $zabbix::params::javagateway_startpollers,
   $timeout                  = $zabbix::params::javagateway_timeout,
 ) inherits zabbix::params {
+  # Fix for pid file. Is different in Zabbix (4, 5) and 6
+  $real_pidfile = $zabbix_version ? {
+    /^[45]\.[024]/ => '/var/run/zabbix/zabbix_java.pid',
+    /^[6]\.[024]/  => $pidfile,
+  }
```

Comment:
This hardcodes the PID file for 4.x and 5.x. Shouldn't the user be able to change the path? 
Maybe we can use the `pick()` function and set the default value of `javagateway_pidfile` to `undef`

```suggestion
  # Fix for pid file. Is different in Zabbix (4, 5) and 6
  $real_pidfile = $zabbix_version ? {
    /^[45]\.[024]/ => pick($pidfile, '/var/run/zabbix/zabbix_java.pid'),
    /^[6]\.[024]/  => pick($pidfile, '/var/run/zabbix/zabbix_java_gateway.pid)
  }
```

Reply:
Yes, agreed. This is a much nicer solution.